File: AEDS-2/TP01/TP01Q08 - Soma de Dígitos/SomaDigitos.java
/*
   ==UserScript==
 @name         TP01Q08 - Soma de Dígitos
 @namespace    https://github.com/Ddavidi/PUC-MINAS
 @description  VERDE PUC MINAS - TP01Q08 - Soma de Dígitos
 @author       @ddavidi_
   ==/UserScript==
*/

import java.util.Scanner;

class SomaDigitos {
    public static boolean isFim(String entrada){
        return entrada.charAt(0) == 'F' && entrada.charAt(1) == 'I' && entrada.charAt(2) == 'M';
    }

    public static int tamanhoEntrada(String entrada){
        return entrada.length();
    }

    public static int somaDigitos(String entrada, int tam, int soma){
        if(tam < 0){
            return soma;
        }
        else{
            soma += Character.getNumericValue(entrada.charAt(tam));
            return somaDigitos(entrada, tam-1, soma);
        }
    }

    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);

        String entrada = scan.nextLine();
        int tam = tamanhoEntrada(entrada);
        
        while(!isFim(entrada)){     
            System.out.println(somaDigitos(entrada, tam-1, 0));
            entrada = scan.nextLine();
            tam = tamanhoEntrada(entrada);
        }
    }
}

File: AEDS-2/TP01/TP01Q22 - RECURSIVO - Soma de Dígitos em C/SomaDigitos.c
/*
   ==UserScript==
 @name         TP01Q22 - RECURSIVO - Soma de Dígitos em C
 @namespace    https://github.com/Ddavidi/PUC-MINAS
 @description  VERDE PUC MINAS - TP01Q22 - RECURSIVO - Soma de Dígitos em C
 @author       @ddavidi_
   ==/UserScript==
*/

#include <stdio.h>

int isEnd(char entrada[]){
    return entrada[0] == 'F' && entrada[1] == 'I' && entrada[2] == 'M' && entrada[3] == '\0';
}

int tamanhoEntrada(char entrada[]){
  int i = 0;

  while (entrada[i] != '\0'){
      i++;
  }

  return i;
}

int somaDigitos(char entrada[], int tam, int soma){
  int charInt;

  if (tam < 0){
    return soma;
  }

  else{
    charInt = entrada[tam] - '0';
    soma += charInt;
    return somaDigitos(entrada, tam-1, soma);
  }
}

int main() {

  char entrada[99];
  int soma;

  scanf(" %[^\r\n]", entrada);

  int tam = tamanhoEntrada(entrada);

  while(!isEnd(entrada)){
    soma = 0;
    printf("%d\n", somaDigitos(entrada, tam-1, soma));
    scanf(" %[^\r\n]", entrada);
    tam = tamanhoEntrada(entrada);
  }

  return 0;
}